localstack/services/stores.py

```diff
@@ -210,18 +214,22 @@ def __getitem__(self, account_id: str) -> RegionBundle:
         if self.validate and not re.match(r"\d{12}", account_id):
             raise ValueError(f"'{account_id}' is not a valid AWS account ID")
 
-        if account_id not in self.keys():
-            self[account_id] = RegionBundle(
-                service_name=self.service_name,
-                store=self.store,
-                account_id=account_id,
-                validate=self.validate,
-            )
+        with BUNDLE_LOCK:
```

Comment:
from my understand this linearizes *all* store access calls on one lock. meaning we can only ever make one concurrent access regardless of whether the bundles are related or not.

would it maybe make sense to add a lock for each bundle instead?

Reply:
Great point @thrau, fixed in 77002c4574743cbe608b74faf45bfd8f6650a873